### PR TITLE
app_voicemail crashes with ODBC backend

### PR DESCRIPTION
**Pull request by sysnux**
Correct 2 bugs in app_voicemail while using ODBC backend.
Resolves: https://github.com/asterisk/asterisk/issues/831

### CODE REVIEW

File: apps/app_voicemail.c [hunk 1]
```diff
@@ -4691,10 +4691,10 @@ struct insert_data {
 
 #define STORE_SQL_FMT_CAT "INSERT INTO %s (dir, msgnum, recording, context, callerid, " \
 	"origtime, duration, mailboxuser, mailboxcontext, flag, msg_id, category) " \
-	"VALUES (?,?,?,?,?,?,?,?,?,?,?,?)"
+	"VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?)"
```
Comment (gtjoseph) [inline apps/app_voicemail.c:4694]: In the master and 21 branches, there are only 12 named parameters but you've added a 13th so this is going to fail.

File: apps/app_voicemail.c [hunk 2]
```diff
@@ -4691,10 +4691,10 @@ struct insert_data {
...
 #define STORE_SQL_FMT "INSERT INTO %s (dir, msgnum, recording, context, callerid, "\
 	"origtime, duration, mailboxuser, mailboxcontext, flag, msg_id) "\
-	"VALUES (?,?,?,?,?,?,?,?,?,?,?)"
+	"VALUES (?,?,?,?,?,?,?,?,?,?,?,?)"
```
Comment (gtjoseph) [inline apps/app_voicemail.c:4697]: Same here.  This statement has 11 named parameters but you add a 12th placeholder.

**Review by gtjoseph — CHANGES_REQUESTED** [review]
First...  You've submitted the PR against the master branch but the issue doesn't actually exist in the master or 21 branches.  Only 20 and 18 will have the issue because they had an extra parameter "macrocontext" which was removed for the 21 and master branches.  You'll need to rebase this PR on the 20 branch instead of master.

Things to do...

- [ ] Sign the Contributor License Agreement.
- [ ] Squash commits into a single one.
- [ ] Make sure the PR description and commit message reference the issue (It's missing the 'Resolves:' before the issue number). See [Commit Messages](https://docs.asterisk.org/Development/Policies-and-Procedures/Commit-Messages/)
- [ ] Add the appropriate "cherry-pick" PR comment. This PR will need to be submitted against the 20 branch and cherry-picked to the 18 branch.  See [Code Contribution](https://docs.asterisk.org/Development/Policies-and-Procedures/Code-Contribution/)

See:
https://docs.asterisk.org/Development/Policies-and-Procedures/Commit-Messages/
https://docs.asterisk.org/Development/Policies-and-Procedures/Coding-Guidelines/
https://docs.asterisk.org/Development/Policies-and-Procedures/Code-Contribution/

File: apps/app_voicemail.c (hunk 1 above)
Comment (sysnux) [reply apps/app_voicemail.c:4694]: Yes, I made a mistake, this only applies to 20 branch.

File: apps/app_voicemail.c (hunk 2 above)
Comment (sysnux) [reply apps/app_voicemail.c:4697]: Yes, I made a mistake, this only applies to 20 branch.

**Comment by gtjoseph** [comment]
@sysnux Are going to resubmit this PR with the changes or should I just take it over?

**Comment by sysnux** [comment]
@gtjoseph Please take it over, thanks.

**Comment by gtjoseph** [comment]
> @gtjoseph Please take it over, thanks.

Will do.  Thanks for finding the issue.

**Comment by gtjoseph** [comment]
Replaced with new PRs.

**Pull request closed without merging**